packages/angular/build/src/utils/i18n-options.ts

```diff
@@ -82,22 +95,41 @@ export function createI18nOptions(
     },
   };
 
-  let rawSourceLocale;
-  let rawSourceLocaleBaseHref;
+  let rawSourceLocale: string | undefined;
+  let rawSourceLocaleBaseHref: string | undefined;
+  let rawsubPath: string | undefined;
   if (typeof metadata.sourceLocale === 'string') {
     rawSourceLocale = metadata.sourceLocale;
   } else if (metadata.sourceLocale !== undefined) {
-    ensureObject(metadata.sourceLocale, 'i18n sourceLocale');
+    ensureObject(metadata.sourceLocale, 'i18n.sourceLocale');
 
     if (metadata.sourceLocale.code !== undefined) {
-      ensureString(metadata.sourceLocale.code, 'i18n sourceLocale code');
+      ensureString(metadata.sourceLocale.code, 'i18n.sourceLocale.code');
       rawSourceLocale = metadata.sourceLocale.code;
     }
 
     if (metadata.sourceLocale.baseHref !== undefined) {
-      ensureString(metadata.sourceLocale.baseHref, 'i18n sourceLocale baseHref');
+      ensureString(metadata.sourceLocale.baseHref, 'i18n.sourceLocale.baseHref');
+      logger?.warn(
+        `The 'baseHref' field under 'i18n.sourceLocale' is deprecated and will be removed in future versions. ` +
+          `Please use 'subPath' instead.\nNote: 'subPath' defines the URL segment for the locale, acting ` +
+          `as both the HTML base HREF and the directory name for output.\nBy default, ` +
+          `if not specified, 'subPath' uses the locale code.`,
+      );
+
       rawSourceLocaleBaseHref = metadata.sourceLocale.baseHref;
     }
+
+    if (metadata.sourceLocale.subPath !== undefined) {
+      ensureValidsubPath(metadata.sourceLocale.subPath, 'i18n.sourceLocale.subPath');
```

Comment:
We should also validate that each sub path is unique.  Otherwise the files will overwrite each other in the output. Maybe track used subPath to locale and then generate an error based on that.

Reply:
Actually probably need to do that after this since a configured sub path could collide with a default value as well.

Reply:
Yes good idea